update build guide (#7167)
diasurgical/DevilutionX: docs/building.md

<details><summary>Windows via MinGW</summary>

> This section is about compiling binary executables for Windows on a Linux system host via [MinGW](https://www.mingw-w64.org/). It's not about compiling for Windows on a Windows host via MinGW. So you first need a Linux system host. If you don't, you can also try WSL (Windows Subsystem for Linux) on Windows.
> 
> If you have any problems with the build process, please refer to the configuration files in the `.github/workflows` folder. These files are automatically executed by the [GitHub Actions](https://docs.github.com/en/actions) service to automate the build. Under the environment described by these configuration files, the compilation is more likely to succeed.

Reviewer: ```suggestion
This section details how to compile a Windows executable using [MinGW](https://www.mingw-w64.org/) on a Linux system, not on a Windows system. If you are using Windows, you can use WSL (Windows Subsystem for Linux) to accomplish this.
```

### Installing dependencies on WSL, Debian and Ubuntu

> The following compilation process has been tested on `ubuntu-22.04`. Success is not guaranteed on Linux distributions other than Ubuntu 22.

Reviewer 2: ```suggestion
The following compilation process has been tested on `ubuntu-22.04`. Success is not guaranteed on Linux distributions other than Ubuntu 22.04
```

Reviewer: @ZhiZe-ZG Ubuntu's version are YY.MM, there is one every 6 month, and a long term every 2 years.

23.10 = October 2023
24.04 LTS = April 2024
24.10 = October 2024
...
26.04 LTS = April 2026

So that's why the minor version (month) is important to include and not just the major (year).

Author: Thanks

### 32-bit

Download the necessary tools and libraries.

```bash

> # Only tested on Ubuntu 22. This won't work on Ubuntu 24 (they are renamed in Ubuntu 24).

Reviewer 2: ```suggestion
# Only tested on Ubuntu 22.04
```
This is still the command for these on 24.04

sudo apt-get install cmake gcc-mingw-w64-i686 g++-mingw-w64-i686 pkg-config-mingw-w64-i686 libz-mingw-w64-dev
```

Download the 32bit MinGW Development Libraries of [SDL2](https://www.libsdl.org/download-2.0.php) and [Libsodium](https://github.com/jedisct1/libsodium/releases) as well as headers for [zlib](https://zlib.net/zlib-1.2.12.tar.gz) and place them in `/usr/i686-w64-mingw32`. This can be done automatically by running `Packaging/windows/mingw-prep.sh`.

### 64-bit

Download the necessary tools and libraries.

```bash

> # Only tested on Ubuntu 22. This won't work on Ubuntu 24 (they are renamed in Ubuntu 24).

Reviewer 2: ```suggestion
# Only tested on Ubuntu 22.04
```
This is still the command for these on 24.04
